src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/TransmissionStatusTelemetryModule.cs

```diff
@@ -0,0 +1,130 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.Extensibility;
+using Microsoft.ApplicationInsights.Extensibility.Implementation;
+using Microsoft.ApplicationInsights.WindowsServer.TelemetryChannel;
+using Newtonsoft.Json;
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.WebJobs.Logging.ApplicationInsights
+{
+    /// <summary>
+    /// Transmission Status Telemetry Module 
+    /// </summary>
+    internal class TransmissionStatusTelemetryModule : ITelemetryModule, IDisposable
+    {
+        private bool isInitialized = false;
+        DiagnosticListener source = new DiagnosticListener(LoggingConstants.HostDiagnosticSourcePrefix + "ApplicationInsights");
+        string debugTracingEventName = LoggingConstants.HostDiagnosticSourceDebugEventNamePrefix + "TransmissionStatus";
+        JsonSerializerSettings options = new JsonSerializerSettings() { Error = (sender, error) => error.ErrorContext.Handled = true, NullValueHandling = NullValueHandling.Ignore };
+
+
+        /// <summary>
+        /// Initializes the telemetry module.
+        /// </summary>
+        /// <param name="configuration">Telemetry configuration to use for initialization.</param>
+        public void Initialize(TelemetryConfiguration configuration)
+        {
+            // Prevent the telemetry module from being initialized multiple times.
+            if (isInitialized)
+            {
+                return;
+            }
+            (configuration.TelemetryChannel as ServerTelemetryChannel).TransmissionStatusEvent += Handler;
+            isInitialized = true;
+        }
+
+        /// <summary>
+        /// Disposes the object.
+        /// </summary>
+        public void Dispose()
+        {
+            source.Dispose();
+        }
+
+        internal void Handler(object sender, TransmissionStatusEventArgs args)
+        {
+            if (args == null)
+            {
+                source.Write("TransmissionStatus", "Unable to parse transmission status response, args is null");
+                return;
+            }
+
+            // Always log if the response is non-success or if the feature flag is enabled
+            if (args.Response?.StatusCode != 200 || source.IsEnabled(debugTracingEventName))
+            {
+                // Do not block the main thread
+                Task.Run(() =>
```

Comment:
I'm not sure doing this in a Task is necessary. The .NET framework and ASP.NET write directly to the source all over the place. Plus it should be rare.